### PR TITLE
CMakeLists.txt changes.

### PR DESCRIPTION
**Pull request by shreejanshrestha**

### CODE REVIEW

**Comment by ctabin** [comment]
Many thanks for your PR. May you just provide me some explanations about your changes ? Besides that, it looks fine to me and will merge it.

**Comment by shreejanshrestha** [comment]
@ctabin 
Sorry for late reply. I didn't notice you replied on my pull request some days ago.
Could you give me some more time for explanations? As I have already forgotten them. I need some to review the commits again to explain.

**Comment by ctabin** [comment]
@shreejanshrestha 
Yes, no problem. Thanks !

File: tests/tests.cpp [hunk 1]
```diff
@@ -42,7 +42,7 @@
 #include <fstream>
 #include <string>
 
-#include "libzippp.h"
```
Comment (ctabin) [inline tests/tests.cpp]: Unfortunately, by changing that, you break the compilation within the Makefile. Can you fix it ?

File: CMakeLists.txt [hunk 2]
```diff
@@ -20,30 +23,32 @@ IF(ZLIB_VERSION_STRING VERSION_LESS "1.1.2")
...
-add_definitions(-DLIBZIPPP_EXPORTS)
 ADD_LIBRARY(libzippp_static STATIC "src/libzippp.cpp")
 
 ADD_EXECUTABLE(libzippp_shared_test "tests/tests.cpp")
 TARGET_LINK_LIBRARIES(libzippp_shared_test libzippp_shared)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MT" FORCE)
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /MTd" FORCE)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MD")
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /MDd")
```
[the remark this answers is not in the document]
Comment (shreejanshrestha) [reply CMakeLists.txt]: MT will have the runtime inside it while, when compiled with MD will link with a DLL at the moment of execution.

File: src/libzippp.h [hunk 3]
```diff
@@ -58,10 +58,9 @@ typedef unsigned int uint;
         typedef long long libzippp_int64;
         typedef unsigned long long libzippp_uint64;
         typedef unsigned short libzippp_uint16;
-        
         //special declarations for windows to use libzippp from a DLL
         #define SHARED_LIBRARY_EXPORT __declspec(dllexport)
-        #define SHARED_LIBRARY_IMPORT __declspec(dllimport)
+        #define SHARED_LIBRARY_IMPORT
```
[the remark this answers is not in the document]
Comment (shreejanshrestha) [reply src/libzippp.h]: No surely not, it won't cause any issue. I have been using this type of sharing same .h file, between shared and static library.

File: tests/tests.cpp (hunk 1 above)
Comment (shreejanshrestha) [reply tests/tests.cpp]: Sorry for that. But I wasn't able to compile without this change, using visual studio solution. By the way I don't know to fix, but could you what type of make system are you using?

**Comment by ctabin** [comment]
Looks fine to me. Thanks for your changes and explanations !

**Pull request merged**